capz/readme.md

```diff
@@ -4,44 +4,96 @@ These scripts and templates are used with [cluster-api-provider-azure](https://g
+export REGISTRY="<registry>"
 
 # optional for ability to use your own ssh key (otherwise it generates one)
 # NOTE: Azure does not support ed25519 encrypted SSH keys!
 export AZURE_SSH_PUBLIC_KEY_FILE="$HOME/.ssh/id_rsa.pub"
 ```
 
-## Other configuration
+#### Optional
 
 | ENV variable  | Description  |
 | ------------- | ------------ |
-| `SKIP_CREATE` | Don't create a cluster.  Must set `CLUSTER_NAME` and have current a workload cluster kubeconfig file with name `./"${CLUSTER_NAME}".kubeconfig` |
-| `SKIP_LOG_COLLECTION` | Don't collect logs from the cluster |
-| `SKIP_TEST`  | Only creates the cluster, will not run tests |
-| `SKIP_CLEANUP` | Don't delete the cluster / resource group after script executions |
+| `API_SERVER_FEATURE_GATES` | Comma-separated list of feature-gates and their values to pass to the kube-apiserver (Defaults to "") |
+| `AZURE_LOCATION` | The azure region to deploy resources into. If not specified a random region will be selected) |
+| `KUBERNETES_VERSION`  | Valid values are `latest` (default) and  `latest-1.xx` where x is valid kubernetes minor version such as `latest-1.24` |
+| `NODE_FEATURE_GATES` | Comma-seperated list of feature-gates and their values to pass to the kubelet (Defaults to "HPAContainerMetrics=true") |
+| `NODE_MACHINE_TYPE` | The [Azure vm size](https://learn.microsoft.com/en-us/azure/virtual-machines/sizes) to use for the nodes  |
 | `RUN_SERIAL_TESTS` | If set to `true` then serial slow tests will be run with default ginkgo settings |
-| `KUBERNETES_VERSION`  | Valid values are `latest` (default) and  `latest-1.x` where x is valid kubernetes minor version such as `latest-1.24` |
-| `AZURE_LOCATION` | The azure region to deploy resources into |
+| `SKIP_CREATE` | Don't create a cluster.  Must set `CLUSTER_NAME` and have current a workload cluster kubeconfig file with name `./"${CLUSTER_NAME}".kubeconfig` |
+| `SKIP_LOG_COLLECTION` | If set to `true` don't collect logs from the cluster |
+| `SKIP_TEST`  | If set to `true` only creates the cluster, will not run tests |
+| `SKIP_CLEANUP` | If set to `true` don't delete the cluster / resource group after script executions |
+| `SCHEDULER_FEATURE_GATES` | Comma-separated list of feature-gates and their values to pass to the kube-scheduler (Defaults to "") |
+| `WINDOWS_CONTAINERD_URL` | URL to a containerd release tarball to use for the Windows nodes (defaults to containerd v1.7.0)|
 | `WINDOWS_KPNG` | If specified, will create a cluster using an out-of-tree kube-proxy implementation from [k-sigs/windows-service-proxy](https://github.com/kubernetes-sigs/windows-service-proxy) |
+| `WINDOWS_SERVER_VERSION` | Set to `windows-2019` (default) or `windows-2022` to test Windows Server 2019 or Windows Server 2022 |
+| `WINDOWS_WORKER_MACHINE_COUNT` | Number of **Windows** worker nodes to provision in the cluster (Defaults to 2) |
+| `WORKER_MACHINE_COUNT` | Number of **Linux** worker nodes to provision in the cluster (Defaults to 2) |
```

Comment:
Does `WORKER_MACHINE_COUNT` have any effect? I am not able to bring up any Linux workers no matter what value I set here. There is `CONTROL_PLANE_MACHINE_COUNT` which is taking effect, however I don't see `NODE_FEATURE_GATES` taking effect for the control plane nodes.

Reply:
oh, the templates we have checked into windows-testing don't have any linux nodes.
I'll remove thisenv var